lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -3114,6 +3114,29 @@ struct CanonicalizeConvertFromHistogram
   }
 };
 
+// The source gather layout does not matter, so we can fold conversions into the
```

Comment:
does not matter because?

I understand it's because it goes through shared memory. Can you explicitly state here?